### DISCUSSION TITLE
File editing in /etc/mail/

### DISCUSSION THREAD

**Discussion by sommerso** [45Drives/cockpit-navigator#28] [Q&A]
Hi, 

I'm want to use navigator as a GUI to edit /etc/mail/relay-domains. This will be an ACL for a jnr Windows Admin to administrate. 

My user is in visudo as well as the wheel group. I've managed to make /etc/mail the default landing page for Navigator and the file has rw-rw-rw permissions, however when I try to save my edit, I get a pop up

Operation not permitted. 

I changed ownership to my user and confirmed that my user can write to the file via the shell. However I'm still having issues with Navigator. 

What am I missing?

**Comment by joshuaboud** [comment]
While trying to recreate this, I noticed that ownership is set to root:root and permissions are reset back to the default 755 whenever I save a file after editing on my machine. Perhaps what you're experiencing is related to this bug. I'll transfer this to an issue and see if fixing that will fix what you're experiencing too.

**Reply by sommerso** [reply]
I have managed to fix this. 

The directory /etc/mail required execution rights. The default rights were 755, I just added execution to that and then chmod 777 to relay-domains within that and I could edit and save the file. 

The execution on mail is required as even chmod 777 results in the "Operation not permitted" 

No sudo or additional permissions required.

**Reply by sommerso** [reply]
Hi Josh, 

It seems what you mentioned is also an issue yes. My testing was done on a single user system. As soon as multiple users work on it, it reverts to ownership of 755 and the last user to have edited the file. The directory still needs to be executable by the group.

Making it impossible to be written by anyone else. 

I've got a script that works around this, but would be nice not to have to do some hackery with file watchers.

**Reply by joshuaboud** [reply]
> Hi Josh,
> 
> It seems what you mentioned is also an issue yes. My testing was done on a single user system. As soon as multiple users work on it, it reverts to ownership of 755 and the last user to have edited the file. The directory still needs to be executable by the group.
> 
> Making it impossible to be written by anyone else.
> 
> I've got a script that works around this, but would be nice not to have to do some hackery with file watchers.

I'm going to start working on this today. Follow #29 for updates.
